bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/ProgressCallbackTest.groovy

```diff
@@ -54,77 +55,112 @@ public final class ProgressCallbackTest {
             getText:  { bundle, key, defaultText, locale, arguments ->
                 usedMessagedKey = key
                 return "Dummy Message"
-            } 
+            }
         ] as I18nProvider
         sut = new ProgressCallbackImpl(new DummyFirmwareHandler(),publisher, i18nProvider, expectedThingUID, expectedFirmwareUID, null)
     }
 
+    @Test(expected=IllegalStateException)
+    void 'assert that update throws IllegalStateException if update is finished'(){
+        sut.defineSequence(ProgressStep.DOWNLOADING, ProgressStep.TRANSFERRING)
+        sut.success()
+        sut.update(100)
+    }
+
+    @Test(expected=IllegalArgumentException)
+    void 'assert that update throws IllegalArgumentException if progress smaller 0'(){
+        sut.defineSequence(ProgressStep.DOWNLOADING, ProgressStep.TRANSFERRING)
+        sut.update(-1)
+    }
+
+    @Test(expected=IllegalArgumentException)
+    void 'assert that update throws IllegalArgumentException if progress greater 100'(){
+        sut.update(101)
+    }
+
+    @Test
+    void 'assert that update results not in FirmwareUpdateProgressInfo event if progress not changed'(){
+        sut.defineSequence(ProgressStep.UPDATING)
+        sut.update(10)
+        sut.update(10)
+        assertThatProgressInfoEventIsValid(postedEvents.get(0), ProgressStep.UPDATING, false, 10)
+        assertThat postedEvents.size(), is(1)
+    }
+
+    @Test
+    void 'assert that update results in FirmwareUpdateProgressInfo event if progress is decreased'(){
```

Comment:
does this make sense? shoulnt we throw a IllegalArgumentException or IllegalStateException if the value is decreased?